File: JS/LeetCode热题HOT100/128_最长连续序列.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var longestConsecutive = function(nums) {
    let set = new Set(nums);
    let ans = 0
    for(let num of set) {
        if(!set.has(num-1)){
            let curLen = 1
            while(set.has(num+1)){
                curLen++
                num++
            }
            ans = Math.max(ans, curLen)
        }
    }
    return ans
};
console.log(longestConsecutive([100,4,200,1,3,2])) // 4 // [1,2,3,4]
console.log(longestConsecutive([0,3,7,2,5,8,4,6,0,1])) // 9 [0,1,2,3,4,5,6,7,8]
// 128. 最长连续序列

File: Python/面试经典150题/哈希表/128_最长连续序列.py
from typing import List
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        if len(nums) == 0:
            return 0
        nums_set = set(nums)
        nums_list = list(nums_set)
        nums_sort = sorted(nums_list)
        ans = 0
        now = 1
        for i in range(1, len(nums_sort)):
            if nums_sort[i] == nums_sort[i-1]+1:
                now += 1
            else:
                ans = max(ans,now)
                now = 1
        return max(ans,now)

nums = [100,4,200,1,3,2]
print(Solution().longestConsecutive(nums))